src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/PackageSpecFactory.cs

```diff
@@ -188,7 +188,7 @@ private static (ProjectRestoreMetadata? RestoreMetadata, List<TargetFrameworkInf
         /// <param name="project">An <see cref="IReadOnlyDictionary{NuGetFramework,ProjectInstance} "/> containing the projects by their target framework.</param>
         /// <param name="isCpvmEnabled">A flag that is true if the Central Package Management was enabled.</param>
         /// <returns>A <see cref="List{TargetFrameworkInformation}" /> containing the target framework information for the specified project.</returns>
-        internal static List<TargetFrameworkInformation> GetTargetFrameworkInfos(IProject project, bool isCpvmEnabled)
+        internal static List<TargetFrameworkInformation> GetTargetFrameworkInfos(IProject project, bool isCpvmEnabled, bool isPruningEnabledGlobally)
```

Comment:
Maybe a comment for param isPruningEnabledGlobally indicating it can be overridden by an MSBuild property?